SHS-6341: Remove modernizr javascript dependency

cienvaras opened the pull request:
# Summary
- [Modernizr is deprecated](https://www.drupal.org/node/3333253), remove it as a theme dependency

## Backstory
- [ClickUp ticket](https://app.clickup.com/t/36718269/SHS-6341)

## Need Review By (Date)
10/08

## Urgency
medium

## Steps to Test
1. No special testing needed, as Modernizr was only used in obsolete browsers. Just confirm that everything works as expected as part of the regular QA.

---
- To see the specific tasks where the Asana app for GitHub is being used, see below:
  - https://app.asana.com/0/0/1211566031442925

cienvaras committed:
chore(shs-6341): remove modernizr dependency

cienvaras commented:
@ahughes3 @Jnwibowo7321 Ready for review. This PR only removes a dependency used in obsolete browsers (e.g. Internet Explorer) so nothing should change. Just confirm that everything works as expected in your regular testing.

Jnwibowo7321 reviewed (approved):
LGTM on Tugboat, dont merge until 10/15 please

joegl committed:
Merge branch '11.18.1-release' into shs-6341--remove-modernizr-theme-dependency

joegl commented:
@cienvaras There is a functional test that keeps failing with this change. I'm not sure if it's related to dropping modernizr or not. Let me know if you want some help investigating.

EDIT: It looks like the arrows next to the menu items are missing, and the `::after` pseudo element is completely gone.

joegl reviewed (changes requested):
See comment about menu issues.

joegl commented:
> @cienvaras There is a functional test that keeps failing with this change. I'm not sure if it's related to dropping modernizr or not. Let me know if you want some help investigating.
> 
> EDIT: It looks like the arrows next to the menu items are missing, and the `::after` pseudo element is completely gone.

@cienvaras I think you can ignore this. The last few hours the same test functional test failures have been showing up across a number of PR's in the codebase with little commonality, and now they seem to be disappearing. It may have to do with some asset caching after the grunt PR (#1995) got merged, especially because the failure was a megamenu issue and the PR changed a few megamenu assets. I'm not entirely sure.

joegl commented:
> > @cienvaras There is a functional test that keeps failing with this change. I'm not sure if it's related to dropping modernizr or not. Let me know if you want some help investigating.
> > EDIT: It looks like the arrows next to the menu items are missing, and the `::after` pseudo element is completely gone.
> 
> @cienvaras I think you can ignore this. The last few hours the same test functional test failures have been showing up across a number of PR's in the codebase with little commonality, and now they seem to be disappearing. It may have to do with some asset caching after the grunt PR (#1995) got merged, especially because the failure was a megamenu issue and the PR changed a few megamenu assets. I'm not entirely sure.

@cienvaras Last follow-up on this. I believe I've resolved the test functional test issue that had cropped up everywhere in #2018. It appears the way the CI workflow was handling the npm steps and assets wasn't fully refreshing the assets properly after the grunt/webpack changes were merged in #1995. Branches and PR that existed pre-merge and then merged in the change were getting test failures as a result of the CI build process.  After I merge #2018 into 11.18.1 I will merge it in here, and that should resolve the test failures. You will also probably want to do the same for any existing PR's.

joegl committed:
Merge branch '11.18.1-release' into shs-6341--remove-modernizr-theme-dependency

joegl commented:
@cienvaras Apologies for all the poking. It does appear that the NPM fix resolved the MegaMenu failure, but there is still a mobile menu test failure potentially being caused by the removal of modernizr.

cienvaras committed:
Merge branch '11.18.1-release' of github.com:SU-HSDO/suhumsci into shs-6341--remove-modernizr-theme-dependency

cienvaras committed:
fix(shs-6341): remove 'no-js' class used by modernizr from html

cienvaras commented:
@joegl Thanks for all the test debugging and the fixes!

This is ready for review again. There was an issue related with the modernizr removal that I missed: there was a hardcoded `no-js` class in the `html` that modernizr removes after execution. Without modernizr, the class was still there and some styles were not working. I removed the class and updated the CSS to match how Drupal works now.
